installer/templates/phx_web/components/core_components.ex

```diff
@@ -35,9 +35,11 @@ defmodule <%= @web_namespace %>.CoreComponents do
         This is another modal.
       </.modal>
 
+  To prevent the modal from accidentally closing when clicking on the backdrop, pass the 'modal' attr.
   """
   attr :id, :string, required: true
   attr :show, :boolean, default: false
+  attr :modal, :boolean, default: false
```

Comment:
I like the idea of conditionally disabling the `phx-click-away`, but I feel the name of the attribute is a bit strange since the component is already called "modal".

Perhaps this was something you struggled with as well?

Brainstorming a bit:

- persistent
- sticky
- noBackdropDismiss
- preventBackdropClose

Reply:
Well, I wouldn't have called it a `modal` of course :-)

Perhaps `blocking` has a similar ring.

Reply:
`confirmClose` maybe?

Reply:
These are separate concerns. Modal behaviour means that interacting with the main content is disabled until the user interacts with the dialog content. That means that clicking on the backdrop does nothing, not even asking for a confirmation.
This type of interaction is common for confirmation dialogs: they're not meant to be closed without clicking OK or Cancel.
In #5660 we agreed that asking for a confirmation provides a better UX for forms in a dialog. Nevertheless there will still be a need to be able to create actual modal dialogs.

Reply:
For accessibility, Escape to close should always be possible.

Reply:
That's the one that trips me the most. So perhaps the best solution is to ask for confirmation indeed. Let's see what Chris has to say after the holidays. :)